app/controllers/api/v2/remote_execution_features_controller.rb

```diff
@@ -28,6 +28,23 @@ def update
         process_response @remote_execution_feature.update_attributes(remote_execution_feature_params)
       end
 
+      def cockpit_url_for_host(host)
+        tmpl = Setting[:remote_execution_cockpit_href]
+        tmpl && tmpl != "" ? tmpl % { :host => host } : nil
+      end
+
+      def cockpit_redirect
+        return head(:forbidden) unless params[:redirect_uri]
+        redir_url = URI.parse(params[:redirect_uri])
+        cockpit_url = cockpit_url_for_host("")
+        redir_url.query = if redir_url.hostname == URI.join(Setting[:foreman_url], cockpit_url).hostname
```

Comment:
we should fail more nicely, when cockpit_url returns nil, probably mentioning to the user they should configure the cockpit url. Currently, one gets ugly error with:

```
bad argument (expected URI object or URI string)
```